resources/assets/js/components/FindSkillsModal.tsx

```diff
@@ -101,6 +120,34 @@ const FindSkillsModal: React.FunctionComponent<FindSkillsModalProps> = ({
   // Used to set the button color of an active skill category.
   const [buttonClicked, setButtonClicked] = useState("");
 
+  /**
+   * Filters through the skills list for the any skills that match the search query.
+   * @param searchQuery The search query entered by the user.
+   */
+  const handleSkillSearch = (searchQuery: string): Promise<void> => {
+    const skillNames: string[] = skills.map((skill) =>
+      localizeFieldNonNull(locale, skill, "name"),
+    );
+    const skillStrings: string[] = matchStringsCaseDiacriticInsensitive(
+      searchQuery,
+      skillNames,
+    );
+    const skillMatches = skills.filter((skill) =>
+      skillStrings.includes(localizeFieldNonNull(locale, skill, "name")),
+    );
+
+    // Set the skillResults state with the matches from the query.
+    setFirstVisit(false);
+    setResultsSectionText({
+      title: intl.formatMessage(messages.searchResultsTitle, {
+        numOfSkills: skillMatches.length,
+        searchQuery,
+      }),
+      description: "",
+    });
+    setSkillsResults(skillMatches);
+    return Promise.resolve();
+  };
   return (
     <section>
```

Comment:
Sorry, something else I noticed after submitting the review - I don't think this component should include a section tag. The parent can do that if it wants.